### PR TITLE
fix: Rewrite Vimeo component to fix a bug and make logic readable

### PR DESCRIPTION
**Pull request by kof**
Closes https://github.com/webstudio-is/webstudio/issues/3097

## Description

1. rewrite logic, hopefully now its readable
2. ensure spinner showing right after click
3. ensure player works even if play button and/or spinner are hidden

## Steps for reproduction

1. @istarkov is aware

## Code Review

- [ ] hi @istarkov , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/sdk-components-react/src/vimeo.tsx [hunk 1]
```diff
@@ -68,7 +62,7 @@ type VimeoPlayerOptions = {
   transparent?: boolean;
 };
 
-const getUrl = (options: VimeoPlayerOptions) => {
+const getVideoUrl = (options: VimeoOptions) => {
```
Comment (istarkov) [inline packages/sdk-components-react/src/vimeo.tsx:65]: Looks like a bug
In UI I see enter video URL, Im entering `https://player.vimeo.com/video/831343124`

The code below transforms it into

```
    const userUrl = new URL(options.url);
    url = new URL(IFRAME_CDN);
    url.pathname = `/video${userUrl.pathname}`;
```

causing double `/video/video` to appear

**Comment by istarkov** [comment]
I think you can warmConnections on useEffect will work on mobile too and it's just `preconnect`

File: packages/sdk-components-react/src/vimeo.tsx [hunk 2]
```diff
@@ -350,6 +224,115 @@ export type VimeoOptions = Omit<
...
+    }
+
+    if (showPreview === false) {
+      onPreviewImageUrlChange(undefined);
+      return;
+    }
+
+    if (previewImageUrl === undefined) {
+      loadPreviewImageUrl(videoUrl)
+        .then(onPreviewImageUrlChange)
```
Comment (istarkov) [inline packages/sdk-components-react/src/vimeo.tsx:291]: Usually this is bug in react as effect can be reexecuted and outdated `onPreviewImageUrlChange` will be called
Any async with callbacks in React are bugs

**Review by istarkov — APPROVED** [review]

File: packages/sdk-components-react/src/vimeo.tsx (hunk 2 above)
Comment (kof) [reply packages/sdk-components-react/src/vimeo.tsx:291]: yeah, potentially, is there an easy way to do fetch from a component?

File: packages/sdk-components-react/src/vimeo.tsx (hunk 1 above)
Comment (kof) [reply packages/sdk-components-react/src/vimeo.tsx:65]: In the UI the expection is to enter vimeo url that you get form the vimeo UI 
image

**Comment by kof** [comment]
> I think you can warmConnections on useEffect will work on mobile too and it's just preconnect
Do you suggest to warm the connection on mobile in useEffect and on desktop on pointer over? Or do you suggest to always  warm in in useEffect

**Pull request merged**